moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -52,7 +52,16 @@ Servo::Servo(ros::NodeHandle& nh, const planning_scene_monitor::PlanningSceneMon
   if (!readParameters())
     exit(EXIT_FAILURE);
 
-  joint_state_subscriber_ = std::make_shared<JointStateSubscriber>(nh_, parameters_.joint_topic);
+  // By default joint topic name is relative to the node parent namespace. E.g. if the joint topic specified in the
+  // configuration files is "joint_states" and the node is launched in namespace "bar", which exists within
+  // namespace "foo", then the subscribed topic would be "/foo/bar/joint_states".
+  //
+  // Fully custom joint name topics can be used by using absolute topic names in config files. For example,
+  // joint topic parameter can be explicitly set to "/baz/my_joint_state_topic".
+  // **Note!** Make sure your planning scene monitor subscribes to correct joint states topic when using absolute
+  // joint state topic name *or* custom relative joint state topic name (for example, "my_joint_states").
+  ros::NodeHandle nh_parent_ns = ros::NodeHandle("");
```

Comment:
Shouldn't this still be a private NodeHandle following the described logic?

Reply:
I may have used wrong terminology, so just to confirm that we are talking about the same thing: If the node would have name `servo_node` and it would exist within namespace `foo`. Then I would argue that the `joint_states` it listens to should be `/foo/joint_states` and not `/foo/servo_node/joint_states`. If the explanation is confusing, can you propose an alternative wording?

Reply:
I see, your solution totally makes sense